src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -33,66 +38,96 @@ public CSharpDeclareAsNullableCodeFixProvider()
+                    if (symbol is IFieldSymbol)
+                    {
+                        // x.field could be used in a test
+                        return true;
+                    }
+                    else if (symbol is IPropertySymbol)
+                    {
+                        // x.Property could be used in a test
+                        return true;
+                    }
+                }
+
+                return false;
+            }
         }
 
-        protected override Task FixAllAsync(
+        protected override async Task FixAllAsync(
             Document document, ImmutableArray<Diagnostic> diagnostics,
             SyntaxEditor editor, CancellationToken cancellationToken)
         {
             // a method can have multiple `return null;` statements, but we should only fix its return type once
             using var _ = PooledHashSet<TypeSyntax>.GetInstance(out var alreadyHandled);
 
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             foreach (var diagnostic in diagnostics)
             {
                 var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
-                MakeDeclarationNullable(editor, node, alreadyHandled);
+                MakeDeclarationNullable(editor, node, alreadyHandled, model);
             }
-
-            return Task.CompletedTask;
         }
 
         protected override bool IncludeDiagnosticDuringFixAll(Diagnostic diagnostic, Document document, string equivalenceKey, CancellationToken cancellationToken)
         {
             var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
-            return equivalenceKey == GetEquivalenceKey(node);
+            var model = document.GetSemanticModelAsync(cancellationToken).GetAwaiter().GetResult();
+            return equivalenceKey == GetEquivalenceKey(node, model);
         }
 
-        private static void MakeDeclarationNullable(SyntaxEditor editor, SyntaxNode node, HashSet<TypeSyntax> alreadyHandled)
+        private static void MakeDeclarationNullable(SyntaxEditor editor, SyntaxNode node, HashSet<TypeSyntax> alreadyHandled, SemanticModel model)
         {
-            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node, model);
             if (declarationTypeToFix != null && alreadyHandled.Add(declarationTypeToFix))
             {
                 var fixedDeclaration = SyntaxFactory.NullableType(declarationTypeToFix.WithoutTrivia()).WithTriviaFrom(declarationTypeToFix);
                 editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
             }
         }
 
-        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
+        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node, SemanticModel model)
```

Comment:
IDE pref is parameters go from wider->narrowe.  So SemanticModels go before nodes within that model.